test/test_intensify/test_intensify.py

```diff
@@ -0,0 +1,351 @@
+from ConfigSpace.hyperparameters import UniformIntegerHyperparameter
+
+from smac.runhistory.runhistory import RunHistory
+from smac.scenario.scenario import Scenario
+from smac.stats.stats import Stats
+from smac.tae.execute_func import ExecuteTAFuncDict
+from smac.intensification.intensification import Intensifier
+from smac.runhistory.runhistory import RunHistory, RunKey
+from smac.smbo.objective import average_cost
+from smac.tae.execute_ta_run import StatusType
+from smac.stats.stats import Stats
+from smac.utils.io.traj_logging import TrajLogger
+
+
+def get_config_space():
+    cs = ConfigurationSpace()
+    cs.add_hyperparameter(UniformIntegerHyperparameter(name='a',
+                                                       lower=0,
+                                                       upper=100))
+    cs.add_hyperparameter(UniformIntegerHyperparameter(name='b',
+                                                       lower=0,
+                                                       upper=100))
+    return cs
+
+
+class TestIntensify(unittest.TestCase):
+
+    def setUp(self):
+        unittest.TestCase.setUp(self)
+
+        self.rh = RunHistory(aggregate_func=average_cost)
+        self.cs = get_config_space()
+        self.config1 = Configuration(self.cs,
+                                     values={'a': 0, 'b': 100})
+        self.config2 = Configuration(self.cs,
+                                     values={'a': 100, 'b': 0})
+        self.config3 = Configuration(self.cs,
+                                     values={'a': 100, 'b': 100})
+
+        self.scen = Scenario({"cutoff_time": 2, 'cs': self.cs,
+                              "output_dir": ""})
+        self.stats = Stats(scenario=self.scen)
+        self.stats.start_timing()
+
+        self.logger = logging.getLogger("Test")
+
+    def test_compare_configs_chall(self):
+        '''
+            challenger is better but has not enough runs
+        '''
+        intensifier = Intensifier(
+            tae_runner=None, stats=self.stats, 
+            traj_logger=TrajLogger(output_dir=None, stats=self.stats), 
+            rng=None,
+            instances=[1])
+
+        self.rh.add(config=self.config1, cost=1, time=2,
+                    status=StatusType.SUCCESS, instance_id=1,
+                    seed=None,
+                    additional_info=None)
+
+        self.rh.add(config=self.config2, cost=1, time=1,
+                    status=StatusType.SUCCESS, instance_id=1,
+                    seed=None,
+                    additional_info=None)
+
+        conf = intensifier._compare_configs(incumbent=self.config1,
+                                            challenger=self.config2,
+                                            run_history=self.rh,
+                                            aggregate_func=average_cost)
+
+        # challenger has enough runs and is better
+        self.assertTrue(conf == self.config2, "conf: %s" % (conf))
```

Comment:
How about `self.assertEqual(conf, self.conf2)`? Or wasn't the output in case of inequality nice to read?